feat: support for post deploy scripts (#1325)
snakemake/snakemake: docs/snakefiles/deployment.rst

.. _singularity:

-------------------------
Providing post-deployment scripts
-------------------------

> With Snakemake 9.14 post-deployment shell-scripts can be provided to perform additional adjustments of a conda environment.

Author: ```suggestion
From Snakemake 6.14 onwards, post-deployment shell-scripts can be provided to perform additional adjustments of a conda environment.
```

This might be helpful in case a conda package is missing components or requires further configuration for execution.

> post-deployment scripts must be placed next to their corresponding environment-file and require the suffix ``.post-deploy.sh``.

Author: ```suggestion
Post-deployment scripts must be placed next to their corresponding environment-file and require the suffix ``.post-deploy.sh``.
```

.. code-block:: python

    rule NAME:
        input:
            "seqs.fastq"
        output:
            "results.tsv"
        conda:
            "envs/interproscan.yaml"

>             "envs/interproscan.post-deploy.sh

Author: ```suggestion
```
The plan was to have them in the same folder, but there is no need to mention them in the conda env definition.

Reviewer: True. My example is wrong. I'll update the documentation.
